gcodeutils/gcode_optimize_arcs.py

```diff
@@ -44,7 +44,7 @@ def main():
     GCodeArcOptimizerFilter().filter(gcode)
 
     # write back modified gcode
-    gcode.write(args.outfile if args.inplace is True else args.outfile)
+    gcode.write(open(args.infile.name, 'w') if args.inplace is True else args.outfile)
```

Comment:
Right. it just won't make sense with both --inplace and the default of sys.stdin, but people looking for stdin/stdout piping don't need --inplace. I'm fine not guard checking it.